pkg/gits/github.go

```diff
@@ -791,25 +791,33 @@ func (p *GitHubProvider) ListOpenPullRequests(owner string, repo string) ([]*Git
 	return answer, nil
 }
 
+func extractRepositoryCommitAuthor(rc *github.RepositoryCommit) (gu *GitUser) {
+	gu = &GitUser{}
+
+	if rc.Commit.Author != nil {
+		gu.Email = rc.Commit.Author.GetEmail()
+		gu.Name = rc.Commit.Author.GetName()
+
+		repositoryAuthorPresent := rc.Author != nil
+		if repositoryAuthorPresent && gu.Email == rc.Author.GetEmail() {
```

Comment:
On second thought. I think this is not quite correct since the origin of this issue was to begin with that `rc.Author.GetEmail()` is quite often empty. AFAIU it will depend on whether you have a public email as part of your profile or not. Many choose not to expose a public email which was leading to the fact that in we went to the file system and the actual git repo to get the email. See `GetPullRequestCommits`. 

With your change, we are in a better position now, since we will always get name and email. But I think it will also lead to the fact that we won't populate these additional fields even though we could/should.

I am wondering whether it would not just be better to just add login, URL and avatar to the `GitUser` provided  `rc.Author != nil`. We basically rely on GitHub to give us a matching Github when possible.  If GitHub cannot match the email it will return a nil Author (the second commit of `curl --user "token:$GH_TOKEN" https://api.github.com/repos/jenkins-x/jx/pulls/5939/commits`, note also that my committer profile does not contain an email, since I don't have a public exposed email).

Reply:
@hferentschik i see what you are saying but I think this could be a problem because at least from the example in that address, we would be adding as email: foo@bar.com and as login and avatar, invalid-email-address and some url respectively, which don't really belong together. What is this information used for later on? because I could see a scenario where login "invalid-email-address" would have lot's of commits related to it, even if they are actually different users.

Reply:
Fair point. I think foo@bar.com is a bit of edge case and I am not sure whether there is actually somehow a "user" with this email. If you really choose random non-existent emails you indeed get empty authors.

I think the most important thing we are after is the name as email as per commit. After that, we need to decide on which side we prefer to error. In your case, we will know that if the `rc.Author.GetEmail` exists and matches, the data will be correct. On the other hand, for all the cases where people decide now to make their email public in their GitHub profile, we won't enhance the `GitUser` at all, even though the data would be there.

In what I am suggesting, we will always enhance the GitUser provided GitHub can make a match, but we might have edge cases like foor@bar.com where you get "invalid" or at the very least questionable data. 

My reasoning is that there will be a proper match in almost all cases. To end up with these border cases someone would have to intentionally use these email addresses. Why would you do that?

Also look at:

```bash
curl --user "token:$GH_TOKEN" https://api.github.com/search/users?q=foo@bar.com
```

I think that there are literally users in the GitHub database using this email address and hence you get this strange matches.

If you use something like:
curl --user "token:$GH_TOKEN" https://api.github.com/search/users?q=john.doe@acme.com
```bash

you get an empty result.

Reply:
@hferentschik sounds good to me, I've applied the changes